app/models/user_access.rb

```diff
@@ -24,10 +39,14 @@ class NotAuthorizedError < StandardError; end
   }.freeze
 
   ACTION_TO_LEVEL = {
-    view: [:manager, :viewer],
+    manage_overdue_list: [:manager, :viewer_all, :call_center],
+    view_reports: [:manager, :viewer_all, :viewer_reports_only],
+    view_pii: [:manager, :viewer_all],
     manage: [:manager]
   }.freeze
 
+  VALID_OPERATIONS = [:access_record, :access_any, :create]
```

Comment:
We've added a new argument called `operation` to the `can?` and `authorize` methods.

This allows us to perform slightly different checks for 
- creating new records (`:create`)
- cases where no explicit record exists (`:access_any`)
- specific records (`:access_record`) 

We tried a version of this where we can infer these cases using `new_record?` and `record == nil`, but this is much cleaner.

See [this](https://github.com/simpledotorg/simple-server/blob/347206af26e89d3da909383ee45a19f676c74b19/app/models/user_access.rb#L62) for the other approach